src/index.ts

```diff
@@ -441,6 +441,64 @@ export default class FireStash extends EventEmitter {
     this.eventsTimer = null;
   }
 
+  async onThrottledSnapshot(
+    documentPath: string,
+    callback: (snapshot: FirebaseFirestore.DocumentSnapshot) => void,
+    timeout = 1000,
+  ): Promise<() => void> {
+    // If we've already started the watcher, return.
+    const pending = this.watchers.get(documentPath);
```

Comment:
Currently, this limits it to one watcher at a time and will no-op if you've already attached a watcher.

Ideally, we'll keep a `Set` of callbacks per documentPath that we iterate over and trigger in order when we get new data.

That'd involve a new property on FireStash eg `throttledSnapshotCallbacks: Map<string, Set<(snapshot: FirebaseFirestore.DocumentSnapshot) => void>>`

We will *always* add callback functions to the set when `onThrottledSnapshot` is called, but we only set up the watchers if it is the first time calling it for that document path (like we do here)

When calling callback() we need to iterate over `this.throttledSnapshotCallbacks.get(documentPath)`

Reply:
Will require one more test to validate multiple callbacks.